### PR TITLE
[2] refactor(approve): replace LegacyConfirmationPendingContent by ConfirmationPendingContent

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

The goal of this PR is removal of `LegacyConfirmationPendingContent` usage.
One of the most difficult places where this component is used is `useTransactionConfirmationModal`.

The problem with `LegacyConfirmationPendingContent` component in the fact that it violates two important SOLID principles: `S` and `D`. The is responsible for displaying many operation types (cancel order, approve, approve via Metamask, etc). It also contains `operationType` prop which violates `D` principle (global entities should not be aware of local entities).

The `LegacyConfirmationPendingContent` must be replaced by `ConfirmationPendingContent` and its special case (e.g `usePendingApprovalModal`).

1. Simplified `<ConfirmationPendingContent>` props, deleted: `statusIcon`, `operationSubmittedMessage`, `walletAddress`, `walletNameLabel`. All those props are resolved inside of the component. 
2. Extracted `MetamaskApproveBanner` as a pure component
3. Added `usePendingApprovalModal` to indicate pending process while approving
4. Introduced the changes above into Tokens page (approving process)

  # To Test

Please, test only through https://github.com/cowprotocol/cowswap/pull/3756

### CODE REVIEW

File: apps/cowswap-frontend/src/common/pure/ConfirmationPendingContent/ConfirmationPendingContent.tsx [hunk 1]
```diff
@@ -1,68 +1,85 @@
...
         <WalletIcon>{statusIcon}</WalletIcon>
         <span>{title}</span>
       </UpperSection>
       <LowerSection>
         <h3>
-          <span>{description}</span>
+          <span>
+            {CustomDescription || (
+              <>
+                <span>{description} </span>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/ConfirmationPendingContent/ConfirmationPendingContent.tsx:52]: Isn't `description` a custom description already?

Would it make sense to turn `description` type into `string|ReactNode` and use it the standard format here only when it's a string?

File: apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <ConfirmationPendingContent
+      onDismiss={onDismissCallback}
+      title={
+        <>
+          Approving <strong>{currencySymbol || context}</strong> for trading
+        </>
+      }
+      description="Approving token"
+      operationLabel="token approval"
+      CustomBody={isMetaMaskDesktop ? <MetamaskApproveBanner /> : undefined}
```
Comment (anxolin) [inline apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx:42]: I was not too aware of this, we want to show the modal only for Metamask? for other wallets, we show something simpler?

File: apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
+import { getIsMetaMask, getWeb3ReactConnection, injectedConnection } from '@cowprotocol/wallet'
+import { useWeb3React } from '@web3-react/core'
+
+import { useModalState } from './useModalState'
+
+import { MediumAndUp, useMediaQuery } from '../../legacy/hooks/useMediaQuery'
+import { ConfirmationPendingContent } from '../pure/ConfirmationPendingContent'
+import { MetamaskApproveBanner } from '../pure/MetamaskApproveBanner'
```
Comment (anxolin) [inline apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx:10]: nit: Probably not worth it, but consider lazy loading if this component is only used in a specific wallet

**Review by anxolin — APPROVED** [review]

File: apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx [hunk 4]
```diff
@@ -148,20 +140,10 @@ export const TokensTableRow = ({
...
 
-  useEffect(() => {
-    if (approvalState === ApprovalState.PENDING) {
-      setApproving(true)
-    } else if (prevApprovalState === ApprovalState.PENDING && approvalState === ApprovalState.NOT_APPROVED) {
-      setApproving(false)
-    }
-  }, [approvalState, prevApprovalState, approving])
+    return <CardsSpinner />
+  }, [currentAllowance, handleApprove, approvalState, theme.green1])
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx:146]: Why does it depend on a theme color?!

File: apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx [hunk 5]
```diff
@@ -104,23 +87,26 @@ export const TokensTableRow = ({
...
+    handleCloseError,
+    handleSetError,
+    toggleWalletModal,
+    tokenData?.symbol,
+    openApproveModal,
+    closeApproveModal,
+  ])
 
   const hasZeroBalance = !balance || balance?.equalTo(0)
-  const hasNoAllowance = !currentAllowance || currentAllowance.equalTo(0)
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx:123]: I assume this behaviour is covered but the approval state?

**Review by alfetopito — APPROVED** [review]
A couple of questions, looks good otherwise

File: apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx:42]: image

Yes, for other wallets we display simplier content

File: apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx (hunk 3 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx:10]: It's too small, don't worth it

File: apps/cowswap-frontend/src/common/pure/ConfirmationPendingContent/ConfirmationPendingContent.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/pure/ConfirmationPendingContent/ConfirmationPendingContent.tsx:52]: Refactored it, thanks!

https://github.com/cowprotocol/cowswap/pull/3756/commits/1c13ea6724ca669627e4f47a49659982c3a2b4d0

File: apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx (hunk 5 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx:123]: Yep, you are right

File: apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx (hunk 4 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx:146]: Good catch!
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/ffa16cd075a10c4e00d5087d81f7250a9f193f61

File: apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx (hunk 2 above)
Comment (anxolin) [reply apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx:42]: ok, makes sense..

**Pull request merged**